### ISSUE TITLE
Support optional nonfree Verilog-A models

### ISSUE THREAD

**Issue by guitorri** [Qucs/qucs-nonfree#2]
Objective: bringing back the non-free model user experience, as an optional extension to QUCS.

on the QUCS end
- [x] implement dynamic loader (dlopen)
- [x] implement a dictionary for components
- [x] make components dynamic, so more can be added with dlopen
- [ ] automatically load (some/selected) installed extensions
- [ ] cleanup the rest of the code, to behave (exactly) as previously
- [ ] provide makefile for symbol module compilation

this repository
- [x] create a stub device for testing (47d5e4a)
- [ ] import verilog code that works with qucs
- [ ] build/install the components for the gui
- [ ] provide binary plugins for qucsator

qucsator
- [x] implement dynamic loader (dlopen)
- [ ] simplify model compilation (revisit makefiles)
- [ ] maybe some minor path issues
- [ ] implement command to build model extentions (`.hdl` or similar)
- [ ] implement command to handle model cards (`.modelcard`, `.include` or similar)

**Comment by felix-salfelder** [comment]
the essentials of the symbol plugins are working (in the component_modules) branch.

you can now do `qucs -a mymodules`, and get the modules from mymodules.so listed in the toolbox. see the resistor for an example. will figure out more of the details...

the idea here is to turn the nonfree modules into `nonfree-modules.so` ...

**Comment by felix-salfelder** [comment]
new idea, question, remark.

the first goal of this endeavour is to reenable the removed functionality. it will be much easier, to put the gui-symbols into `qucs/contrib/legacy_va_symbols` (is there anything non-free about them?). this will facilitate the gui-symbol-interface development in the early phase. also it will enable us to focus on the *final* we-attach-custom-models-and-symbols, rather than reimplement legacy functionality from the outside.

tl;dr; put the nonfree va code here. and put the old symbols/graphics into the gui repo, to be loaded optionally.

what do you think?
